src/core/mac/sub_mac.cpp

```diff
@@ -487,7 +498,12 @@ void SubMac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, Error aErro
         {
             mCallbacks.RecordCcaStatus(ccaSuccess, aFrame.GetChannel());
         }
-
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+        if (mCslPeriod > 0)
+        {
+            mCslLastSync = TimeMicro(static_cast<uint32_t>(aAckFrame->mInfo.mRxInfo.mTimestamp));
```

Comment:
I think this timestamp is the timestamp of the end of received ACK, not the timestamp of the beginning of the frame being transmitted. To get correct number, you should subtract ACK duration, AIFS, and the transmitted frame duration.

Reply:
Yes. Worst case should be: 2*(127+41)*16+192 = 5568 us. This imprecision will only add 1 us to the receive window after more than 200 s from last synchronization, assuming ±20 ppm crystals in both parent and child. Do we need extra arithmetic to cope with this small error?

Reply:
It actually 2 * (127+41)***32**+192 = 10944 us. For worst case: 200s assuming RC +/- 500 PPM, we get 5.5 us error. I think we don't need to bother with such small error, because it is much smaller than scheduling granularity. However, I would add a comment that we intentionally omit this error.

Reply:
Timestamp from `aAckFrame` can't be used since not all platforms provide it (including simulation). Getting back to current MAC time and adding a comment assuming the small error.

Reply:
This assumption may be incorrect on some platforms. I'm fine to keep it right now to move forward.
But I would prefer to use radio timestamps on platforms where this feature is available, to avoid potential problems caused by software latency.